### PR TITLE
Use the worker API for Gradle 5.6 and above

### PR DESCRIPTION
**Pull request by lacasseio**
It fixes https://github.com/jbake-org/jbake-gradle-plugin/issues/56 and https://github.com/jbake-org/jbake-gradle-plugin/issues/53 on Gradle 5.6 and above. For Gradle 5.5 and lower, nothing has changed. I created a fix for Gradle 5.5 and lower but it presented a regression: https://github.com/lacasseio/jbake-gradle-plugin/commit/37044a4fff25f595029b3e5ec5829caede29dcca.

Before:
Screen Shot 2020-06-27 at 9 54 47 AM

After:
Screen Shot 2020-06-27 at 9 54 56 AM

### CODE REVIEW

File: src/main/groovy/org/jbake/gradle/impl/JBakeWorkActionParameters.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.gradle.api.provider.Property;
+import org.gradle.workers.WorkParameters;
+
+interface JBakeWorkActionParameters extends WorkParameters {
+    RegularFileProperty getInput()
+    RegularFileProperty getOutput()
+
+    Property<Boolean> getClearCache()
+
+    MapProperty<String, Object> getConfiguration();
```
Comment (aalmiray) [inline src/main/groovy/org/jbake/gradle/impl/JBakeWorkActionParameters.groovy:31]: Semicolon can be removed

Comment (lacasseio) [reply src/main/groovy/org/jbake/gradle/impl/JBakeWorkActionParameters.groovy]: Done.

**Comment by lacasseio** [comment]
> Don't forget to add and `@author` tag to the files you've created.

Regarding the `@author` tag, I don't really see the value given the git history will give the same information. I added them but you can cherry-pick out the commit.

**Comment by lacasseio** [comment]
I will push another commit to remove the comment as coveralls fails to not consider them during the coverage calculation. I don't have time to start playing with those tools, especially when the change at hand has no impact on the result. Feel free to cherry-pick or change anything. The patch is completed and works as expected.

**Comment by lacasseio** [comment]
I'm not sure I understand how coveralls work. The code is covered by the tests already present. Is there any more action on my part to get the PR merge and release?

**Comment by lacasseio** [comment]
Thanks, the condition is already covered in a test that try the minimum and latest Gradle (pre and post 5.6). It would have been nice to remove the condition but it breaks pre 5.6 as the logs aren't piped properly to the daemon so I left the old code in.

**Comment by lacasseio** [comment]
Any chance this gets release soon? I don't want to seem too pushy but it does slow down the development of my documentation portal and I'm starting to contemplate just forking the project which would be counter-productive.

**Pull request merged**